### PR TITLE
xcur2png: fix cross-build

### PR DESCRIPTION
**Pull request by fufexan**
## Description of changes

Fixes errors caused by autoconf not finding rpl_malloc. Fix implemented by following the answer [here](https://www.linuxquestions.org/questions/linux-software-2/undefined-reference-to-%60rpl_malloc%27-587256/).

```
/nix/store/vl7wj0kj6rq3lgn7czb266lbr5y26kfn-aarch64-unknown-linux-gnu-binutils-2.42/bin/aarch64-unknown-linux-gnu-ld: /build/source/xcur2png.c:230:(.text+0x91c): undefined reference to `rpl_malloc'
```

Tested with `nix build .#pkgsCross.aarch64-multiplatform.xcur2png` and `nix build >#pkgsCross.riscv64.xcur2png`.

Also add patch fixing wrong math, thanks @Artturin for the suggestion.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fufexan** [comment]
Created a PR upstream with the patch at https://github.com/eworm-de/xcur2png/pull/3.

File: pkgs/tools/graphics/xcur2png/default.nix [hunk 1]
```diff
@@ -11,6 +19,18 @@ stdenv.mkDerivation rec {
     sha256 = "0858wn2p14bxpv9lvaz2bz1rk6zk0g8zgxf8iy595m8fqv4q2fya";
   };
 
+  patches = [
+    # implement malloc workaround to allow xcur2png to be cross-compiled
+    # https://www.linuxquestions.org/questions/linux-software-2/undefined-reference-to-%60rpl_malloc%27-587256/
+    ./malloc.diff
```
Comment (Artturin) [inline pkgs/tools/graphics/xcur2png/default.nix:25]:
```suggestion
    # https://github.com/eworm-de/xcur2png/pull/3
    ./malloc.diff
```

Perhaps also submit the other patch? No idea why arch maintainers didn't do that.

Comment (fufexan) [reply pkgs/tools/graphics/xcur2png/default.nix:25]: Looks like the commit already exists on a separate branch https://github.com/eworm-de/xcur2png/commit/aa035462d950fab35d322cb87fd2f0d702251e82. Should we fetch the patch from there?

Comment (Artturin) [reply pkgs/tools/graphics/xcur2png/default.nix:25]: > Looks like the commit already exists on a separate branch [eworm-de/xcur2png@aa03546](https://github.com/eworm-de/xcur2png/commit/aa035462d950fab35d322cb87fd2f0d702251e82). Should we fetch the patch from there?

Yes

**Pull request merged**